Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -416,27 +417,44 @@ final class PurchasesOrchestrator {
                   promotionalOffer: PromotionalOffer.SignedData? = nil,
                   winBackOffer: WinBackOffer? = nil,
                   metadata: [String: String]? = nil,
+                  trackDiagnostics: Bool,
                   completion: @escaping PurchaseCompletedBlock) {
         Self.logPurchase(product: product, package: package, offer: promotionalOffer)
 
+        self.trackPurchaseStartedIfNeeded(trackDiagnostics: trackDiagnostics,
+                                          productId: product.productIdentifier,
+                                          productType: product.productType)
+        let startTime = self.dateProvider.now()
+
+        let completionWithTracking: PurchaseCompletedBlock =
+        { [weak self] transaction, customerInfo, error, userCancelled in
```

Comment:
Ohh I haven't done the `[weak self]` in my completion blocks 😬. Will need to do that, good thinking!

Reply:
It's probably not a huge deal since this is mostly a singleton... But we should avoid unnecessary leaks in case the SDK is configured multiple times.

Reply:
Exactly. Not a big deal but I try to do it if it makes sense to avoid leaks, just in case